responder/events.go

```diff
@@ -0,0 +1,109 @@
+package responder
+
+import (
+	"fmt"
+
+	"github.com/flanksource/commons/logger"
+	_ "github.com/flanksource/kommons"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/flanksource/incident-commander/responder/jira"
+)
+
+func ReconcileEvents(events []api.Event) {
+
+	var responderIDs []string
+	responderIDEventMap := make(map[string]api.Event)
+	for _, event := range events {
+		responderIDs = append(responderIDs, event.Properties["id"])
+		responderIDEventMap[event.Properties["id"]] = event
+	}
+
+	var responders []api.Responder
+
+	tx := db.Gorm.Find(&responders).Where("id IN (?) AND external_id is NULL", responderIDs).Preload("Team")
+	if tx.Error != nil {
+		logger.Errorf("Error while fetching responders from database: %v", tx.Error)
+		return
+	}
+
+	var externalID string
+	var err error
+	for _, responder := range responders {
+		event := responderIDEventMap[responder.ID.String()]
+		if responder.Properties["responderType"] == "Jira" {
+			fmt.Println("Inside responder properties jira")
+			externalID, err = handleJiraResponder(responder)
+			if err != nil {
+				setErr := event.SetErrorMessage(err.Error())
+				if setErr != nil {
+					logger.Errorf("Error updating table:event_queue with id:%s and error:%s", event.ID, err)
+				}
+				continue
+			}
+		}
+
```

Comment:
refactor this into into a notifyResponder function annd leave behind in the responder package